apps/examples/nettest/nettest.c

```diff
@@ -199,64 +262,75 @@ void ipmcast_sender_thread(int num_packets, uint32_t sleep_time)
 	 * The IP address specified must be associated with a local,
 	 * multicast-capable interface.
 	 */
-	localInterface.s_addr = inet_addr(LOCAL_DEVICE);
+	ret = netlib_get_ipv4addr(intf, &localInterface);
+	if (ret == -1) {
+		printf("[MCASTCLIENT] fail to get interface's ip address\n");
+		goto errout_with_socket;
+	}
+	printf("[MCASTCLIENT] bind interface(%s)\n", intf);
+	printf("[MCASTCLIENT] group address(%s)\n", g_app_target_addr);
+	printf("[MCASTCLIENT] port(%d)\n", g_app_target_port);
+
 	if (setsockopt(sd, IPPROTO_IP, IP_MULTICAST_IF, (char *)&localInterface, sizeof(localInterface)) < 0) {
-		printf("\n[MCASTCLIENT] [ERR] Failed setting local interface");
+		printf("[MCASTCLIENT] [ERR] Failed setting local interface");
 		goto errout_with_socket;
 	}
-	printf("\n[MCASTCLIENT] setsockopt IP_MULTICAST_IF success\n");
+	printf("[MCASTCLIENT] setsockopt IP_MULTICAST_IF success\n");
 
 	/*
 	 * Send a message to the multicast group specified by the
 	 * groupSock sockaddr structure.
 	 */
-
 	datalen = strlen(databuf);
-	printf("\n[MCASTCLIENT] datalen = %d \n", datalen);
-
+	printf("[MCASTCLIENT] datalen = %d \n", datalen);
+	printf("[MCASTCLIENT] data (%s)\n", databuf);
+	printf("[MCASTCLIENT] interval(%d)\n", sleep_time);
 	for (i = 1; i <= num_packets; i++) {
-		printf("\n[MCASTCLIENT] sending mcast message (%s) length (%d) number (%d)\n", databuf, datalen, i);
+		printf("[MCASTCLIENT] sending mcast message length (%d) number (%d)\n", datalen, i);
 		ret = sendto(sd, databuf, datalen, 0, (struct sockaddr *)&groupSock, addrlen);
 		if (ret < 0) {
-			printf("\n[MCASTCLIENT] [ERR] sending datagram message");
+			printf("[MCASTCLIENT] [ERR] sending datagram message");
+		}
+		if (sleep_time != 0) {
+			usleep(sleep_time);
 		}
-		sleep(1);
 	}
 
-	printf("\n[MCASTCLIENT] Terminate multicast sender after sending sufficient messages (%d)\n", num_packets);
+	printf("[MCASTCLIENT] Terminate multicast sender after sending sufficient messages (%d)\n", num_packets);
 	close(sd);
 	return;
 
 errout_with_socket:
 	close(sd);
```

Comment:
Please, remove these two lines and move 'errout_with_socket' between line 299 and line 300.

Reply:
ok